webapp/app/routes/app/proposals.$id/_route.tsx

```diff
@@ -163,6 +165,35 @@ export default function Proposals() {
       >
         <Await resolve={asyncData}>
           {({ addresses, proposal, reports, userSignedLegalVeto, userSignedProposal }) => {
+            const securityCouncilSignaturesReached =
+              proposal.signatures.approveUpgradeSecurityCouncil.length >=
+              NECESSARY_SECURITY_COUNCIL_SIGNATURES;
+            const guardiansSignaturesReached =
+              proposal.signatures.approveUpgradeGuardians.length >= NECESSARY_GUARDIAN_SIGNATURES;
+
+            const signProposalEnabled =
+              !userSignedProposal &&
+              proposal.status === PROPOSAL_STATES.Waiting &&
+              (user.role === "guardian" ? !proposal.guardiansApproval : true);
+            const signLegalVetoEnabled =
+              !userSignedLegalVeto &&
+              !proposal.guardiansExtendedLegalVeto &&
+              proposal.status === PROPOSAL_STATES.LegalVetoPeriod &&
+              user.role === "guardian";
+
+            const executeSecurityCouncilApprovalEnabled =
+              securityCouncilSignaturesReached && user.role === "securityCouncil";
+            const executeGuardiansApprovalEnabled =
+              guardiansSignaturesReached && user.role === "guardian";
```

Comment:
I belive that the veto period have to have ended. The easiest way, I believe, is checking that the state is "Waiting"

Reply:
Fixed!